evm/contracts/apps/ucs/03-zkgm/Send.sol

```diff
@@ -217,78 +212,57 @@ contract UCS03ZkgmSendImpl is Versioned, UCS03ZkgmStore {
         }
     }
 
-    function _verifyFungibleAssetOrderV2(
+    function _verifyTokenOrderV2(
         uint32 channelId,
         uint256 path,
-        FungibleAssetOrderV2 calldata order
+        TokenOrderV2 calldata order
     ) internal {
         address baseToken = address(bytes20(order.baseToken));
```

Comment:
I suggest changing the implementation of this function to switch/match on TokenOrderV2.kind on the toplevel for three reasons:

1. easier to read and follow what happens for each kind
2. more gas-efficient (no need to calculate wrappings)
3. slightly less error prone